module.go

```diff
@@ -0,0 +1,142 @@
+package caddy_libp2p_listener
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/caddyserver/caddy/v2"
+	"github.com/libp2p/go-libp2p"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/routing"
+	libp2phttp "github.com/libp2p/go-libp2p/p2p/http"
+	"github.com/libp2p/go-libp2p/p2p/net/gostream"
+	libp2pwebrtc "github.com/libp2p/go-libp2p/p2p/transport/webrtc"
+	"github.com/multiformats/go-multiaddr"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func init() {
+	//caddy.RegisterModule(new(LL))
+	caddy.RegisterNetwork("multiaddr:", registerMultiaddrURI)
```

Comment:
@MarcoPolo this (along with the hack to add a `/` to the returned address) seems to allow passing multiaddr URIs. A bit gross, but seems ok